widget/table.go

```diff
@@ -0,0 +1,299 @@
+}
+
+func (t *tableRenderer) MinSize() fyne.Size {
+	return t.cellSize
+}
+
+func (t *tableRenderer) Refresh() {
+	template := t.t.NewCell()
+	t.cellSize = template.MinSize().Add(fyne.NewSize(theme.Padding()*2, theme.Padding()*2))
+	t.moveOverlay()
+
+	t.colMarker.FillColor = theme.PrimaryColor()
+	t.colMarker.Refresh()
+	t.rowMarker.FillColor = theme.PrimaryColor()
+	t.rowMarker.Refresh()
+
+	for _, div := range t.dividers {
+		div.(*canvas.Rectangle).FillColor = theme.ShadowColor()
+		div.Refresh()
+	}
+	t.t.cells.Refresh()
+}
+
+func (t *tableRenderer) BackgroundColor() color.Color {
+	return theme.BackgroundColor()
+}
+
+func (t *tableRenderer) Objects() []fyne.CanvasObject {
+	return t.objects
+}
+
+func (t *tableRenderer) Destroy() {
+}
+
+type tableCells struct {
+	BaseWidget
+	t        *Table
+	cellSize fyne.Size
+}
+
+func newTableCells(t *Table, s fyne.Size) *tableCells {
+	c := &tableCells{t: t, cellSize: s}
+	c.ExtendBaseWidget(c)
+	return c
+}
+
+func (c *tableCells) CreateRenderer() fyne.WidgetRenderer {
+	return &tableCellsRenderer{cells: c, pool: &syncPool{}, objects: []fyne.CanvasObject{}}
+}
+
+func (c *tableCells) Tapped(e *fyne.PointEvent) {
+	if e.Position.X < 0 || e.Position.X >= c.Size().Width || e.Position.Y < 0 || e.Position.Y >= c.Size().Height {
+		c.t.SelectedColumn = -1
+		c.t.SelectedRow = -1
+		c.t.Refresh()
+		return
+	}
+
+	c.t.SelectedColumn = e.Position.X / (c.cellSize.Width + 1)
+	c.t.SelectedRow = e.Position.Y / (c.cellSize.Height + 1)
+
+	if c.t.OnCellSelected != nil {
+		c.t.OnCellSelected(c.t.SelectedRow, c.t.SelectedColumn)
+	}
+
+	if c.t.scroll != nil {
+		c.t.scroll.onOffsetChanged()
+	}
+}
+
+type tableCellsRenderer struct {
+	cells   *tableCells
+	pool    pool
+	objects []fyne.CanvasObject
+}
+
+func (r *tableCellsRenderer) Layout(s fyne.Size) {
+	rows, cols := r.cells.t.DataSize()
+
+	// TODO only visible
+	if len(r.objects) == 0 {
+		r.ensureCells()
+	}
+	i := 0
+	for y := 0; y < rows; y++ {
+		for x := 0; x < cols; x++ {
+			c := r.objects[i]
+			c.Move(fyne.NewPos(theme.Padding()+x*r.cells.cellSize.Width+(x-1), theme.Padding()+y*r.cells.cellSize.Height+(y-1)))
+			c.Resize(r.cells.cellSize)
+			i++
+		}
+	}
+}
+
+func (r *tableCellsRenderer) MinSize() fyne.Size {
+	rows, cols := r.cells.t.DataSize()
+	return fyne.NewSize(r.cells.cellSize.Width*cols+(cols-1), r.cells.cellSize.Height*rows+(rows-1))
+}
+
+func (r *tableCellsRenderer) Refresh() {
+	template := r.cells.t.NewCell()
+	r.cells.cellSize = template.MinSize().Add(fyne.NewSize(theme.Padding()*2, theme.Padding()*2))
+	r.ensureCells() // TODO force refresh, once caching doesn't reset them all each time :)
+}
+
+func (r *tableCellsRenderer) BackgroundColor() color.Color {
+	return theme.BackgroundColor()
+}
+
+func (r *tableCellsRenderer) Objects() []fyne.CanvasObject {
+	return r.objects
+}
+
+func (r *tableCellsRenderer) Destroy() {
+	r.clearPool()
+}
+
+func (r *tableCellsRenderer) clearPool() {
```

Comment:
clearPool is a misleading name - it is not clearing the pool, rather it is releasing all cells into the pool

Reply:
Resolved I think